### ISSUE TITLE
libicalvcal seems to leak memory

### ISSUE THREAD

**Issue by bosim** [libical/libical#283]
I have the following program:

```
#include "libical/icalvcal.h"
#include "libical/vcc.h"

int main() {
        const char *x = "Test";
        const char *y = "Test2";

        VObject *root = newVObject(VCCardProp);
        VObject *node, *tmpnode;
        node = addGroup(root, VCNameProp);
        tmpnode = addProp(node, VCGivenNameProp);
        setVObjectStringZValue(tmpnode, x);
        tmpnode = addProp(node, VCFamilyNameProp);
        setVObjectStringZValue(tmpnode, y);

        cleanVObject(root);
}
```

At exit valgrind reports:

```==29096== Memcheck, a memory error detector
==29096== Copyright (C) 2002-2015, and GNU GPL'd, by Julian Seward et al.
==29096== Using Valgrind-3.12.0 and LibVEX; rerun with -h for copyright info
==29096== Command: ./a.out
==29096== 
==29096== 
==29096== HEAP SUMMARY:
==29096==     in use at exit: 26 bytes in 2 blocks
==29096==   total heap usage: 14 allocs, 12 frees, 279 bytes allocated
==29096== 
==29096== 26 (24 direct, 2 indirect) bytes in 1 blocks are definitely lost in loss record 2 of 2
==29096==    at 0x4C29110: malloc (in /usr/lib64/valgrind/vgpreload_memcheck-amd64-linux.so)
==29096==    by 0x4E3E2F3: newStrItem (vobject.c:152)
==29096==    by 0x4E3E2F3: lookupStr (vobject.c:646)
==29096==    by 0x4E3E328: newVObject (vobject.c:118)
==29096==    by 0x4E3E34B: addProp (vobject.c:304)
==29096==    by 0x400890: main (in /home/bo/tmp/a.out)
==29096== 
==29096== LEAK SUMMARY:
==29096==    definitely lost: 24 bytes in 1 blocks
==29096==    indirectly lost: 2 bytes in 1 blocks
==29096==      possibly lost: 0 bytes in 0 blocks
==29096==    still reachable: 0 bytes in 0 blocks
==29096==         suppressed: 0 bytes in 0 blocks
==29096== 
==29096== For counts of detected and suppressed errors, rerun with: -v
==29096== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: 0 from 0)
```

So when using groups, it seems like libicalvcal is leaking memory, with respect to output it looks correct.

I am using libical from openSUSE LEAP 42.2 (1.0.1-12.1)

**Comment by winterz** [comment]
verified in master. I'll take a look

**Comment by winterz** [comment]
fixed in 352da5249229e8a4139c7fe54eb0e7df6397f144 master
